vehicle/OVMS.V3/components/vehicle_mgev/src/mg_can_handler.cpp

```diff
@@ -89,7 +89,7 @@ void OvmsVehicleMgEv::IncomingPollFrame(CAN_frame_t* frame)
         // }
     }
 
-    uint8_t frameType = frame->data.u8[0] >> 4;
+    uint8_t frameType = frame->data.u8[1] >> 4;
```

Comment:
I'm not sure that this change is correct

Reply:
I was wary of that as well. From my understanding, the first byte is supposed to represent data length right? (The line below it also suggests this: 
`uint8_t frameLength = frame->data.u8[0] & 0x0f;`
)
Are there any other cases where this wouldn't be the case?

Reply:
The bottom 4 bits of the first byte are the frame length, the top four bits are the frame type.

Reply:
Oh I see. I think I might have confused frameType with the service ID in the second byte. So in that case, instead of checking `(frameType & 4)` for the gateway and BCM authentication response, we should check `data[0] & 4` instead then?